### PR TITLE
DIP-311 custom templates

### PR DESCRIPTION
**Pull request by cameron-white-janrain**
```
$ ./src/main.js
Usage: src/main.js [input] [options]

Options:
  --bare, -b       Omits top level HTML elements and styles if true.
                                                                [default: false]
  --postmanId, -p  Add Postman Collection ID                    [default: false]

Custom Templates:
  --template-documentation    Path to custom documentation template (overwrites)
                                                                        [string]
  --template-index            path to custom index template (overwrites)[string]
  --template-before-main      path to custom before main template       [string]
  --template-main             path to custom main template (overwrites) [string]
  --template-after-main       path to custom after main template        [string]
  --template-parameters       path to custom parameters template (overwrites)
                                                                        [string]
  --template-resource         path to custom resource template (overwrites)
                                                                        [string]
  --template-securityScheme   path to custom security scheme template
                              (overwrites)                              [string]
  --template-tableOfContents  path to custom table of contents template
                              (overwrites)                              [string]

Custom Styles:
  --before-style  Custom before style sheet (LESS)                      [string]
  --style         Custom style sheet (LESS) (overwrites)                [string]
  --after-style   Custom after style sheet (LESS)                       [string]

Examples:
  src/main.js myDocs.raml > myDocs.html
  src/main.js myDocs.raml -b > myDocs.html

RAML file required
```

### CODE REVIEW

**Comment by cameron-white-janrain** [comment]
Replacement for #40

**Comment by cameron-white-janrain** [comment]
I added `template-after-tableOfContents`

```
$ ./src/main.js
Usage: src/main.js [input] [options]

Options:
  --bare, -b       Omits top level HTML elements and styles if true.
                                                                [default: false]
  --postmanId, -p  Add Postman Collection ID                    [default: false]

Custom Templates:
  --template-documentation          Path to custom documentation template
                                    (overwrites)                        [string]
  --template-index                  path to custom index template (overwrites)
                                                                        [string]
  --template-before-main            path to custom before main template [string]
  --template-main                   path to custom main template (overwrites)
                                                                        [string]
  --template-after-main             path to custom after main template  [string]
  --template-parameters             path to custom parameters template
                                    (overwrites)                        [string]
  --template-resource               path to custom resource template
                                    (overwrites)                        [string]
  --template-securityScheme         path to custom security scheme template
                                    (overwrites)                        [string]
  --template-tableOfContents        path to custom table of contents template
                                    (overwrites)                        [string]
  --template-after-tableOfContents  path to custom after table of contents
                                    template                            [string]

Custom Styles:
  --before-style  Custom before style sheet (LESS)                      [string]
  --style         Custom style sheet (LESS) (overwrites)                [string]
  --after-style   Custom after style sheet (LESS)                       [string]

Examples:
  src/main.js myDocs.raml > myDocs.html
  src/main.js myDocs.raml -b > myDocs.html

RAML file required
```

**Comment by BCox-Janrain** [comment]
There's a lot of customization added to the templates. The code generally looks good. :+1:

File: package.json [hunk 1]
```diff
@@ -1,17 +1,17 @@
 {
   "name": "raml-fleece",
-  "version": "0.6.1",
+  "version": "0.7.0",
```
Comment (ebeesonjanrain) [inline package.json]: the version should not be changed in a feature branch, but in a release branch or commit

Comment (cameron-white-janrain) [inline package.json]: Ok. Ill remove the commit.

**Comment by ebeesonjanrain** [comment]
looks good

**Pull request merged**